cli/cmd/upgrade.go

```diff
@@ -252,6 +257,15 @@ func upgrade(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, stage
 		}
 	}
 
+	// If values is still nil, then neither the linkerd-config-overrides secret
+	// nor the legacy values were found. This means either means that Linkerd
+	// was installed with Helm or that the installation needs to be repaired.
+	if values == nil {
+		return bytes.Buffer{}, errors.New("Could not find the Linkerd config. If Linkerd was installed with Helm, " +
+			"please use Helm to perform upgrades. If Linkerd was not installed with Helm, please use the " +
+			"`linkerd repair` command to repair the Linkerd config")
```

Comment:
This would be clearer to me if it explicitly referred to a resource like `secret/linkerd-config-overrides` instead of "the Linkerd config"

Reply:
Also, this should be wrapped at 80 characters.

Reply:
this is complicated by the fact that for backwards compatibility, we try loading the config from either secret/linkerd-config-overrides or from configmap/linkerd-config.  Not sure that this error message is the right place to explain all of that.